provider/schema/database.go

```diff
@@ -159,3 +213,45 @@ func TruncateTableConstraint(name string) string {
 	}
 	return name
 }
+
+func getResourceValues(r *Resource) ([]interface{}, error) {
+	values := make([]interface{}, 0)
+	for _, c := range append(r.table.Columns, GetDefaultSDKColumns()...) {
+		v := r.Get(c.Name)
+		if err := c.ValidateType(v); err != nil {
+			return nil, err
+		}
+		values = append(values, v)
+		if c.Type == TypeJSON {
+			if reflect2.TypeOf(v).Kind() == reflect.Map {
+				values = append(values, v)
+				continue
+			}
+			switch data := v.(type) {
```

Comment:
The downside of this is that we have to keep track of all the value types that could be used inside a database, am I right with that assumption? I don't see another way to solve this so I guess it is fine but it might be worth keep this in our heads for the future.

Reply:
Yes you are right, but its more related to pgx and how it encodes data from what I saw its mostly only json that is required to be a map[ type